Xamarin.Forms.Platform.Android/Material/MaterialButtonRenderer.cs

```diff
@@ -444,87 +286,139 @@ void UpdateBitmap()
-			{
+			if (_disposed || Element == null)
 				return;
-			}
-
-			Font font = Button.Font;
 
-			if (font == Font.Default && _defaultFontSize == 0f)
-			{
+			Font font = Element.Font;
+			if (font == Font.Default && _defaultTypeface == null && _defaultFontSize < 0f)
 				return;
-			}
 
-			if (_defaultFontSize == 0f)
-			{
+			if (_defaultTypeface == null)
 				_defaultTypeface = Typeface;
-				_defaultFontSize = TextSize;
-			}
 
-			if (font == Font.Default)
-			{
+			if (Element.Font == Font.Default)
 				Typeface = _defaultTypeface;
-				SetTextSize(ComplexUnitType.Px, _defaultFontSize);
-			}
 			else
-			{
 				Typeface = font.ToTypeface();
-				SetTextSize(ComplexUnitType.Sp, font.ToScaledPixel());
-			}
-		}
 
-		void UpdateInputTransparent()
-		{
-			if (Element == null || _isDisposed)
-			{
-				return;
-			}
+			if (_defaultFontSize < 0f)
+				_defaultFontSize = TextSize;
 
-			_inputTransparent = Element.InputTransparent;
+			if (Element.FontSize < 0f)
+				SetTextSize(ComplexUnitType.Px, _defaultFontSize);
+			else
+				SetTextSize(ComplexUnitType.Sp, font.ToScaledPixel());
 		}
 
 		void UpdateText()
 		{
-			if (Element == null || _isDisposed)
-			{
+			if (_disposed || Element == null)
 				return;
-			}
 
 			string oldText = Text;
-			Text = Button.Text;
+			Text = Element.Text;
 
 			// If we went from or to having no text, we need to update the image position
-			if (IsNullOrEmpty(oldText) != IsNullOrEmpty(Text))
+			if (string.IsNullOrEmpty(oldText) != string.IsNullOrEmpty(Text))
 			{
-				UpdateBitmap();
+				UpdateImage();
 			}
 		}
 
 		void UpdateTextColor()
 		{
-			if (Element == null || _isDisposed || _textColorSwitcher == null)
+			if (_disposed || Element == null || _textColorSwitcher == null)
 				return;
 
-			if (Button.TextColor == Color.Default)
-				_textColorSwitcher.Value.UpdateTextColor(this, Button.TextColor);
-			else
-				_textColorSwitcher.Value.UpdateTextColor(this, Color.White);
+			_textColorSwitcher.Value.UpdateTextColor(this, Element.TextColor);
 		}
 
 		void UpdatePadding()
 		{
 			if (Element.IsSet(Button.PaddingProperty))
+			{
 				SetPadding(
-					(int)(Context.ToPixels(Button.Padding.Left) + _paddingDeltaPix.Left),
-					(int)(Context.ToPixels(Button.Padding.Top) + _paddingDeltaPix.Top),
-					(int)(Context.ToPixels(Button.Padding.Right) + _paddingDeltaPix.Right),
-					(int)(Context.ToPixels(Button.Padding.Bottom) + _paddingDeltaPix.Bottom)
-				);
+					(int)(Context.ToPixels(Element.Padding.Left) + _paddingDeltaPix.Left),
+					(int)(Context.ToPixels(Element.Padding.Top) + _paddingDeltaPix.Top),
+					(int)(Context.ToPixels(Element.Padding.Right) + _paddingDeltaPix.Right),
+					(int)(Context.ToPixels(Element.Padding.Bottom) + _paddingDeltaPix.Bottom));
+			}
+		}
+
+		void UpdateBorder()
+		{
+			if (_disposed || Element == null)
+				return;
+
+			var cornerRadius = Element.CornerRadius;
+			if (cornerRadius >= 0f || _defaultCornerRadius >= 0f)
```

Comment:
As with most of the properties, first check to see if we are setting default values to defaults. This reduces the number of times we actually talk via JNI. And, no need to set the same values.